.github/workflows/release.yml

```diff
@@ -38,36 +26,162 @@ jobs:
+          cache-key: '${{ matrix.job.target }}'
+
+      - name: Build Xline Binary
+        run: |
+          cross build --target ${{ matrix.job.target }} --release
+
+      - name: Upload Asset
         id: upload-release-asset
-        uses: actions/upload-release-asset@v1.0.1
+        uses: actions/upload-release-asset@v1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          upload_url: ${{ steps.create_release.outputs.upload_url }}
-          asset_path: ./target/release/xline
-          asset_name: xline-linux-amd64
+          upload_url: ${{ needs.create_release.outputs.upload_url }}
+          asset_path: ./target/${{ matrix.job.target }}/release/xline
+          asset_name: xline-${{ matrix.job.target }}
           asset_content_type: application/octet-stream
-      - run: |
-          mv ./target/release/xline ./scripts
-          mv ./target/release/benchmark ./scripts
+
+      # Build docker image across multiple runners
+      - name: Move Cross-compiled Binary
+        if: startsWith(matrix.job.platform, 'linux/')
+        run: |
+          mv ./target/${{ matrix.job.target }}/release/xline ./scripts
+          mv ./target/${{ matrix.job.target }}/release/benchmark ./scripts
+
       - name: Set up QEMU
+        if: startsWith(matrix.job.platform, 'linux/')
         uses: docker/setup-qemu-action@v1
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
+
       - name: Login to DockerHub
-        uses: docker/login-action@v1
+        if: startsWith(matrix.job.platform, 'linux/')
+        uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Generate App Version
-        run: echo APP_VERSION=`git describe --tags --always` >> $GITHUB_ENV
-      - name: Build and push
-        id: docker_build
-        uses: docker/build-push-action@v2
+
+      - name: Build Docker Image
+        if: startsWith(matrix.job.platform, 'linux/')
+        id: build_image
+        uses: docker/build-push-action@v4
         with:
-          push: true
           context: ./scripts
-          tags: |
-            datenlord/xline:latest
-            datenlord/xline:${{ env.APP_VERSION }}
+          platforms: ${{ matrix.job.platform }}
+          outputs: type=image,name=${{ env.REGISTRY_IMAGE }},push-by-digest=true,name-canonical=true,push=true
+
+      - name: Export Digest
+        if: startsWith(matrix.job.platform, 'linux/')
+        run: |
+          mkdir -p /tmp/digests
+          digest="${{ steps.build_image.outputs.digest }}"
+          touch "/tmp/digests/${digest#sha256:}"
+
+      - name: Upload Digest
+        uses: actions/upload-artifact@v3
+        if: startsWith(matrix.job.platform, 'linux/')
+        with:
+          name: digests
+          path: /tmp/digests/*
+          if-no-files-found: error
+          retention-days: 1
+
+  merge_image:
```

Comment:
This `merge_image` is identical to the one in docker.yml. This is a kind of duplication. Can we extract it into a separate yaml file and reuse it?

Reply:
If we are using something like a [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) to separate these steps, we may need to checkout the xline repository once again to get this `<action>.yml` file, while in `merge_image` job we have no reason to checkout xline repository.